### PR TITLE
Teleop

### PR DESCRIPTION
**Pull request by jake161**
Just finished implementation, will test before completing pull request

### CODE REVIEW

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp [hunk 1]
```diff
@@ -58,60 +58,57 @@ void setup() {
...
         // Append the character to the command string
         command += incomingChar;
       }
     }
 
-    if(daTwist.linear_z == 0.0) {
-      //Note: AlternatePin must be set to LOW before setting the other pin to HIGH! Both pins can never both be high
+    if (daTwist.linear_z == 0.0) {
+      // Note: AlternatePin must be set to LOW before setting the other pin to
+      // HIGH! Both pins can never both be high
```
Comment (SrMeissel) [inline src/arduino_serial_package/arduino_files/linac/src/linac.cpp:81]: I'd leave "HIGH!" on the first line.

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp [hunk 2]
```diff
@@ -58,60 +58,57 @@ void setup() {
...
+    if (daTwist.angular_y == 0) {
+      digitalWrite(backwardPin_top, LOW);
+      digitalWrite(forwardPin_top, LOW);
+
+    } else if (daTwist.angular_y > 0.0) {
+      digitalWrite(backwardPin_top, LOW);
+      digitalWrite(forwardPin_top, HIGH);
+    } else if (daTwist.angular_y < 0.0) {
+      digitalWrite(forwardPin_top, LOW);
+      digitalWrite(backwardPin_top, HIGH);
```
Comment (SrMeissel) [inline src/arduino_serial_package/arduino_files/linac/src/linac.cpp:104]: inconsistent formatting

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp [hunk 3]
```diff
@@ -141,7 +138,8 @@ void loop() {
 //   hallB_old = hallB;
 
 //     // Check encoder publishing conditions
-//     if (millis() - encoderTimer > encoderPeriod && encoderCount_old != encoderCount) {
+//     if (millis() - encoderTimer > encoderPeriod && encoderCount_old !=
+//     encoderCount) {
```
Comment (SrMeissel) [inline src/arduino_serial_package/arduino_files/linac/src/linac.cpp:142]: another new line? in a comment?

**Review by SrMeissel — CHANGES_REQUESTED** [review]
Check comments. simple syntax/formatting questions

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp [hunk 4]
```diff
@@ -58,60 +58,57 @@ void setup() {
...
+      digitalWrite(forwardPin_top, LOW);
+      digitalWrite(backwardPin_top, HIGH);
+    }
+  }
 }
 
 //   // Check encoder publishing conditions
-//   if (millis() - encoderTimer > encoderPeriod && encoderCount_old != encoderCount) {
+//   if (millis() - encoderTimer > encoderPeriod && encoderCount_old !=
+//   encoderCount) {
```
[the remark this answers is not in the document]
Comment (jake161) [reply src/arduino_serial_package/arduino_files/linac/src/linac.cpp:111]: Fixed

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp (hunk 2 above)
Comment (jake161) [reply src/arduino_serial_package/arduino_files/linac/src/linac.cpp:104]: Fixed

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp (hunk 1 above)
Comment (jake161) [reply src/arduino_serial_package/arduino_files/linac/src/linac.cpp:81]: Fixed

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp [hunk 5]
```diff
@@ -58,60 +58,57 @@ void setup() {
...
   // Check if serial data is available
   if (Serial.available()) {
-     while (Serial.available() > 0) // Check if any characters are available
+    while (Serial.available() > 0) // Check if any characters are available
     {
       char incomingChar = Serial.read(); // Read the incoming character
 
-      if (incomingChar == '/') // Delimiter, check if it's the end of the message
+      if (incomingChar ==
+          '/') // Delimiter, check if it's the end of the message
```
[the remark this answers is not in the document]
Comment (jake161) [reply src/arduino_serial_package/arduino_files/linac/src/linac.cpp:66]: Fixed

File: src/arduino_serial_package/arduino_files/linac/src/linac.cpp [hunk 6]
```diff
@@ -55,51 +55,51 @@ void setup() {
...
       digitalWrite(forwardPin_bottom, LOW);
       digitalWrite(backwardPin_bottom, HIGH);
     }
-    if (daTwist.angular_y == 0) {
+    if (daTwist.angular_y == 0){
       digitalWrite(backwardPin_top, LOW);
       digitalWrite(forwardPin_top, LOW);
 
-    } else if (daTwist.angular_y > 0.0) {
+    }
```
Comment (SrMeissel) [inline src/arduino_serial_package/arduino_files/linac/src/linac.cpp:97]: what the hell man

**Review by SrMeissel — APPROVED** [review]

**Pull request merged**